src/SoapClient.php

```diff
@@ -520,4 +509,34 @@ public function handleAsyncResult($functionName, $result)
 
         return $this->__call($functionName, array());
     }
+
+    /**
+     * Returns the headers set for this client object.
+     * 
+     * @return array
+     */
+    public function getHeaders()
+    {
+        return $this->_headers;
+    }
+
+    /**
+     * Returns the service name
+     * 
+     * @return array
```

Comment:
This one is a string, same for the endpoint URL

Reply:
Updated, and created an issue for the type hinting. https://github.com/softlayer/softlayer-api-php-client/issues/30